reporter/osm.py

```diff
@@ -127,21 +127,183 @@ def fetch_osm(file_path, url_path):
         raise
 
 
-def add_keyword_timestamp(keywords_file_path):
-    """Add the current date / time to the keywords file.
+def add_metadata_timestamp(metadata_file_path):
+    """Add the current date / time to the metadata file.
 
-    :param keywords_file_path: Keywords file path that the timestamp should be
+    :param metadata_file_path: Metadata file path that the timestamp should be
         written to.
-    :type keywords_file_path: str
+    :type metadata_file_path: str
     """
     time_stamp = time.strftime('%d-%m-%Y %H:%M')
-    keywords_file = file(keywords_file_path, 'ab')
-    keywords_file.write('date: %s' % time_stamp)
-    keywords_file.close()
+
+    extension = os.path.splitext(metadata_file_path)[1]
+
+    if extension == 'keywords':
+        keyword_file = file(metadata_file_path, 'ab')
+        keyword_file.write('date: %s' % time_stamp)
+        keyword_file.close()
+    else:
+        # Need to write this section : write date/time in XML file
+        pass
+
+
+def get_shorter_version(version):
+    """Get a shorter version, only with the major and minor version.
+
+    :param version: The version.
+    :type version: str
+
+    :return 'major.minor' version number.
+    :rtype float
+    """
+    return float('.'.join(version.split('.')[0:2]))
+
+
+def get_resource_path(feature_type):
+    """Get the resource folder according to the OSM feature.
+
+    :param feature_type: The type of feature.
+    :type feature_type: str
+
+    :return The resource folder.
+    :rtype str
+    """
+    return os.path.abspath(
+        os.path.join(
+            os.path.dirname(__file__),
+            'resources',
+            RESOURCES_MAP[feature_type]))
+
+
+def latest_xml_metadata_file(feature):
+    """Get the latest version available of the XML metadata for a feature.
+
+    :param feature: The feature.
```

Comment:
Better to use the opportunity to explain what feature is....by reading this method alone I wouldn't know...